psi4/driver/driver.py

```diff
@@ -689,23 +699,19 @@ def gradient(name, **kwargs):
         if opt_iter == 1:
             print('Performing finite difference calculations')
 
-        # Shifting the geometry so need to copy the active molecule
-        moleculeclone = molecule.clone()
-
         # Obtain list of displacements
-        findif_meta_dict = driver_findif.gradient_from_energy_geometries(moleculeclone)
+        findif_meta_dict = driver_findif.gradient_from_energy_geometries(molecule)
         ndisp = len(findif_meta_dict["displacements"]) + 1
 
-        # This version is pretty dependent on the reference geometry being last (as it is now)
         print(""" %d displacements needed ...""" % (ndisp), end='')
 
-        for n, displacement in enumerate(findif_meta_dict["displacements"].values(), start=1):
-            _process_displacement(
-                energy, lowername, moleculeclone, displacement, n, ndisp, write_orbitals=False, **kwargs)
-
-        wfn = _process_displacement(energy, lowername, moleculeclone, findif_meta_dict["reference"], ndisp, ndisp,
+        wfn = _process_displacement(energy, lowername, molecule, findif_meta_dict["reference"], 1, ndisp,
```

Comment:
Can you elaborate on the new looping structure here?

Reply:
This isn't a new looping structure, just a change in the order in which we compute displacements. The reference displacement, previously done last, is now moved to be first. The argument to `_process_displacement` specifying the displacement number for print purposes has been changed accordingly.

The point of the change is so the reference is guaranteed to be available for use as an orbital guess, per https://github.com/psi4/psi4/issues/1296#issuecomment-431018568.